patroni/ha.py

```diff
@@ -1206,28 +1221,34 @@ def process_manual_failover_from_leader(self) -> Optional[str]:
 
         :returns: action message if demote was initiated, None if no action was taken"""
         failover = self.cluster.failover
+        # if there is no failover key or
+        # I am holding the lock but am not primary = I am the standby leader,
+        # then do nothing
         if not failover or (self.is_paused() and not self.state_handler.is_primary()):
             return
 
+        action = self._get_failover_action_name()
+
+        # it is not the time for the the scheduled failover yet, do nothing
         if (failover.scheduled_at and not
-            self.should_run_scheduled_action("failover", failover.scheduled_at, lambda:
+            self.should_run_scheduled_action(action, failover.scheduled_at, lambda:
```

Comment:
I guess this line will look strange: https://github.com/zalando/patroni/blob/82d2ef4878cd327e63fdb25e6faaf47dca3cebbe/patroni/ha.py#L1203